### PR TITLE
Add SupplyChain CLI and python client to the Sawtooth REST API

### PR DESCRIPTION
**Pull request by cianx**
Signed-off-by: Cian Montgomery <cian.montgomery@intel.com>

### CODE REVIEW

File: families/supplychain/python/sawtooth_supplychain/cli/record.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    subparsers.add_parser('list')
+
+
+def add_record_show_parser(subparsers):
+    parser = subparsers.add_parser('show')
+
+    parser.add_argument(
+        'name',
+        type=str,
+        help='an identifier for the new game')
```
Comment (agunde406) [inline families/supplychain/python/sawtooth_supplychain/cli/record.py]: Update Help to match actually use.

File: families/supplychain/python/tests/test_supplychain_cli.py [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+        # see that the configuration is cleaned up.
+        self.cli(['reset'])
+        self.expect_no_files(config_files)
+
+    # the following parameters are:
+    # 1) function to patch
+    # 2) cli arguments
+    # 3) return value from the patched function
+    # 4) expected parameters to the patched function.
+    # NOTE: the traling commas in the single item tuples are intential. Python
```
Comment (agunde406) [inline families/supplychain/python/tests/test_supplychain_cli.py]: traling -> trailing

File: families/supplychain/python/tests/test_supplychain_cli.py [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+    # requires this, () with out commas are treated as grouping operators no
+    # as a tuple constructor.
```
Comment (agunde406) [inline families/supplychain/python/tests/test_supplychain_cli.py]: no -> not?

File: families/supplychain/python/tests/test_supplychain_integration.py [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+                     app_status=Application.OPEN):
+        return Application(
+            record_identifier=record_id,
+            applicant=public_key,
+            creation_time=creation_time,
+            type=Application.OWNER,
+            status=Application.OPEN
+        )
+
+    def test_agent(self):
```
Comment (agunde406) [inline families/supplychain/python/tests/test_supplychain_integration.py]: Please add Doc strings to the rest of theses tests.

File: families/supplychain/python/sawtooth_supplychain/cli/agent.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+def add_agent_list_parser(subparsers):
+    subparsers.add_parser(
+        'list',
+        help='Show a list of all the Agents')
+
+
+def add_agent_show_parser(subparsers):
+    parser = subparsers.add_parser(
+        'show',
+        help='Show an Agents')
```
Comment (agunde406) [inline families/supplychain/python/sawtooth_supplychain/cli/agent.py]: Agents -> Agent

File: families/supplychain/python/sawtooth_supplychain/cli/reset.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+import os
+import shutil
+
+
+def add_reset_parser(subparsers, parent_parser):
+    subparsers.add_parser('reset', parents=[parent_parser])
+
+
+def do_reset(args, config):
```
Comment (jsmitchell) [inline families/supplychain/python/sawtooth_supplychain/cli/reset.py]: Eek. Should a `supplychain reset` really delete the users ~/.sawtooth directory?

Comment (cianx) [inline families/supplychain/python/sawtooth_supplychain/cli/reset.py]: Copied this from xo, thought it was the approved method of doing a reset? Is there a better pattern to follow?

Comment (cianx) [inline families/supplychain/python/sawtooth_supplychain/cli/reset.py]: https://github.com/hyperledger/sawtooth-core/blob/master/sdk/examples/xo_python/sawtooth_xo/xo_cli.py#L218

Comment (cianx) [inline families/supplychain/python/sawtooth_supplychain/cli/reset.py]: @jsmitchell ^

Comment (jsmitchell) [inline families/supplychain/python/sawtooth_supplychain/cli/reset.py]: Yeah, seems ill-advised there too. @aludvik

**Pull request merged**